src/diff/component.js

```diff
@@ -62,7 +62,7 @@ export function renderComponent(
 		c = internal._component;
 	} else {
 		// Instantiate the new component
-		if ('prototype' in type && type.prototype.render) {
+		if (internal._flags & TYPE_CLASS) {
```

Comment:
Could we leave this line as-is for now? I have another PR where the `TYPE_CLASS` flag gets removed as part of the "make our internals more agnostic to component implementation" work

Reply:
Definitely - I couldn't generate any meaningful metrics to support this change when I tried, best to just leave it as-is.